src/main/groovy/io/openliberty/tools/gradle/tasks/DevTask.groovy

```diff
@@ -0,0 +1,507 @@
+import org.gradle.api.tasks.options.Option
+import org.gradle.tooling.BuildException;
+import org.gradle.tooling.BuildLauncher
+import org.gradle.tooling.ProjectConnection
+import org.gradle.tooling.GradleConnector
+
+import java.util.concurrent.ArrayBlockingQueue
+import java.util.concurrent.ThreadPoolExecutor;
+
+import io.openliberty.tools.ant.ServerTask
+
+import io.openliberty.tools.common.plugins.util.DevUtil
+import io.openliberty.tools.common.plugins.util.PluginExecutionException
+import io.openliberty.tools.common.plugins.util.PluginScenarioException
+import io.openliberty.tools.common.plugins.util.ServerFeatureUtil
+import io.openliberty.tools.common.plugins.util.ServerStatusUtil;
+
+
+import java.util.concurrent.TimeUnit
+
+class DevTask extends AbstractServerTask {
+
+    DevTask() {
+        configure({
+            description 'Runs a Liberty server in dev mode'
+            logging.level = LogLevel.INFO
+            group 'Liberty'
+        })
+    }
+
+    DevTaskUtil util = null;
+
+    private boolean hotTests = false;
+
+    @Option(option = 'hotTests', description = 'If set to true, run tests automatically after every change. The default value is false.')
+    void setHotTests(boolean hotTests) {
+        this.hotTests = hotTests;
+    }
+
+    private boolean skipTests = false;
+
+    @Option(option = 'skipTests', description = 'If set to true, do not run any tests in dev mode. The default value is false.')
+    void setSkipTests(boolean skipTests) {
+        this.skipTests = skipTests;
+    }
+
+    boolean libertyDebug = true;
+
+    @Option(option = 'libertyDebug', description = 'Whether to allow attaching a debugger to the running server. The default value is true.')
+    void setLibertyDebug(String libertyDebug) {
+        this.libertyDebug = Boolean.parseBoolean(libertyDebug)
+    }
+
+    int libertyDebugPort = 7777;
+
+    @Option(option = 'debugPort', description = 'The debug port that you can attach a debugger to. The default value is 7777.')
+    void setLibertyDebugPort(String libertyDebugPort) {
+        try {
+            this.libertyDebugPort = libertyDebugPort.toInteger();
+        } catch (NumberFormatException e) {
+            logger.error(String.format("Unexpected value: %s for dev mode option libertyDebugPort. libertyDebugPort should be a valid integer.", libertyDebugPort));
+            throw e;
+        }
+    }
+
+    private double compileWait = 0.5;
+
+    @Option(option = 'compileWait', description = 'Time in seconds to wait before processing Java changes and deletions. The default value is 0.5 seconds.')
+    void setCompileWait(String compileWait) {
+        try {
+            this.compileWait = compileWait.toDouble();
+        } catch (NumberFormatException e) {
+            logger.error(String.format("Unexpected value: %s for dev mode option compileWait. compileWait should be a valid number.", compileWait));
+            throw e;
+        }
+    }
+
+    private int verifyTimeout = 30;
```

Comment:
Do some of these options map to properties in `ServerExtension`? Do we (or would it make sense to) use those if they get set in build.gradle?

Reply:
Changed to use CLI args if provided, else use the `ServerExtension` properties, else use the defaults values.